includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -597,6 +596,8 @@ private function process_node( $node ) {
 					$is_bind_attribute = (
 						'[' === $name[0]
 						||
+						AMP_DOM_Utils::AMP_BIND_DATA_ATTR_PREFIX === substr( $name, 0, 40 )
```

Comment:
Shouldn't that be `14` instead of `40`?

Reply:
Yes it should! But it turned out to still pass the unit test because the next condition did pass:

```php
( isset( $this->rev_alternate_attr_name_lookup[ $name ] ) && '[' === $this->rev_alternate_attr_name_lookup[ $name ][0] )
```

Where `$this->rev_alternate_attr_name_lookup` maps all `data-amp-bind-*` attributes to their bracketed versions. All of that is just a waste now, because otherwise that array should just contain:

```json
{
    "srcset": "src",
    "href": "xlink:href"
}
```

But as it stands right now it has 110 other entries for redundant `data-amp-bind-*` mappings. I'm going to eliminate it along with my proposed follow-ups.